src/clients/python/experimental_api_v2/library/grpcclient.py

```diff
@@ -0,0 +1,674 @@
+            an appropriately sized batch of inputs.
+        request_id: string
+            Optional identifier for the request which will be
+            returned in the response. Default value is '0'.
+        sequence_id : int
+            The sequence ID of the inference request. Default is 0, which
+            indicates that the request is not part of a sequence. The
+            sequence ID is used to indicate that two or more inference
+            requests are in the same sequence.
+
+        Returns
+        -------
+        InferResult
+            The object holding the result of the inference, including the
+            statistics.
+
+        Raises
+        ------
+        InferenceServerException
+            If server fails to perform inference.
+        """
+
+        self._request = grpc_service_v2_pb2.ModelInferRequest()
+        self._request.model_name = model_name
+        self._request.model_version = model_version
+        self._request.id = request_id
+        self._request.sequence_id = sequence_id
+        for infer_input in inputs:
+            self._request.inputs.extend([infer_input._get_tensor()])
+        for infer_output in outputs:
+            self._request.outputs.extend([infer_output._get_tensor()])
+
+        try:
+            self._response = self._client_stub.ModelInfer(self._request)
+            self._result = InferResult(self._response)
+            return self._result
+        except grpc.RpcError as rpc_error:
+            raise_error_grpc(rpc_error)
+
+    def async_infer(self,
+                    callback,
+                    inputs,
+                    outputs,
+                    model_name,
+                    model_version=-1,
+                    batch_size=1,
+                    flags=0,
+                    sequence_id=0):
+        """Run inference using the supplied 'inputs' to obtain the outputs
+        specified by 'outputs'.
+
+        Parameters
+        ----------
+        callback : function
+            Python function that is invoked once the request is completed.
+            The function must reserve the last argument to hold InferResult
+            object which will be provided to the function when executing
+            the callback. The ownership of this InferResult object will be
+            given to the user and the its lifetime is limited to the scope
+            of this function.
+        inputs : list
+            A list of InferInput objects, each describing data for a input
+            tensor required by the model.
+        outputs : list
+            A list of InferOutput objects, each describing how the output
+            data must be returned. Only the output tensors present in the
+            list will be requested from the server.
+        batch_size : int
+            The batch size of the inference. Each input must provide
+            an appropriately sized batch of inputs.
+        flags : int
```

Comment:
Flags here but not above, and no request_id here. The flags should be the same on infer and async_infer (except callback).